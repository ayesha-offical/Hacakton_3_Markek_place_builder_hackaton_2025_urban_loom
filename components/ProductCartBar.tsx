"use client"
import React from 'react'
import {MdFavoriteBorder} from "react-icons/md"
import { Button } from './ui/button'
import { cn } from '@/lib/utils'
import toast from 'react-hot-toast'


const ProductCartBar = () => {
  const handleAddtoWishList = () => {
    // Add the logic to add the product to the cart
    // You can use the provided product object as a reference
    toast.success("Added In Wish List");
  };
  return (
    <div className='text-gray-500 w-[170%] text-lg flex items-center justify-center gap-2.5
    '>
      
    <Button 
        onClick={handleAddtoWishList}
    
    className={cn('border mb-2 w-6 h-7 shadow-md bg-black  text-white hover:bg-white hover:text-black hoverEffect rounded-full ')}
    aria-label="Add to Wishlist">
<MdFavoriteBorder/>
</Button>




    </div>
  )
}

export default ProductCartBar
