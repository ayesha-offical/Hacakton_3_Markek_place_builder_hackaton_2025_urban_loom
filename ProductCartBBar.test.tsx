import { render, screen, fireEvent } from "@testing-library/react";
import ProductCartBar from "@/components/ProductCartBar"; // Adjust import path if necessary
import toast from "react-hot-toast";
import "@testing-library/jest-dom";

// Mock react-hot-toast to track toast.success calls
jest.mock("react-hot-toast", () => ({
  success: jest.fn(),
}));

describe("ProductCartBar Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly and matches snapshot", () => {
    const { container } = render(<ProductCartBar />);
    
    // Check if the button with aria-label "Add to Wishlist" is rendered
    expect(screen.getByRole("button", { name: /add to wishlist/i })).toBeInTheDocument();
    
    // Snapshot test to verify the rendered output
    expect(container).toMatchSnapshot();
  });

  test("calls toast.success when the 'Add to Wishlist' button is clicked", () => {
    render(<ProductCartBar />);
    
    const button = screen.getByRole("button", { name: /add to wishlist/i });
    fireEvent.click(button);
    
    // Ensure toast.success is called with the expected message
    expect(toast.success).toHaveBeenCalledWith("Added In Wish List");
  });
});
