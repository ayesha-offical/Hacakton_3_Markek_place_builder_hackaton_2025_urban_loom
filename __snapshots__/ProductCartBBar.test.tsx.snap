// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders correctly with 0 items when groupedItems returns empty array 1`] = `
<div>
  <a
    class="flex items-center md:text-sm gap-2 border border-gray-200 px-2 py-1 rounded-md shadow-md hover:shadow-none text-xs "
    href="/cart"
  >
    <svg
      class="lucide lucide-shopping-cart text-cyan-950 w-5 h-5 md:w-6 md:h-6 hoverEffect"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="21"
        r="1"
      />
      <circle
        cx="19"
        cy="21"
        r="1"
      />
      <path
        d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
      />
    </svg>
    <div
      class="flex flex-col "
    >
      <p
        class="text-xs"
      >
        <span
          class="font-semibold "
        >
          0
        </span>
        items
      </p>
      <p
        class="font-semibold md:text-base text-xs"
      >
        Cart
      </p>
    </div>
  </a>
</div>
`;

exports[`Cart Component renders correctly with correct item count when groupedItems returns items 1`] = `
<div>
  <a
    class="flex items-center md:text-sm gap-2 border border-gray-200 px-2 py-1 rounded-md shadow-md hover:shadow-none text-xs "
    href="/cart"
  >
    <svg
      class="lucide lucide-shopping-cart text-cyan-950 w-5 h-5 md:w-6 md:h-6 hoverEffect"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="21"
        r="1"
      />
      <circle
        cx="19"
        cy="21"
        r="1"
      />
      <path
        d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
      />
    </svg>
    <div
      class="flex flex-col "
    >
      <p
        class="text-xs"
      >
        <span
          class="font-semibold "
        >
          3
        </span>
        items
      </p>
      <p
        class="font-semibold md:text-base text-xs"
      >
        Cart
      </p>
    </div>
  </a>
</div>
`;

exports[`ProductCartBar Component renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="text-gray-500 w-[170%] text-lg flex items-center justify-center gap-2.5   "
  >
    <button
      aria-label="Add to Wishlist"
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 px-4 py-2 border mb-2 w-6 h-7 shadow-md bg-black text-white hover:bg-white hover:text-black hoverEffect rounded-full"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
    </button>
  </div>
</div>
`;
